homeassistant/components/sky_hub/device_tracker.py

```diff
@@ -1,63 +1,71 @@
 """Support for Sky Hub."""
 import logging
-import re
 
-import requests
+from pyskyqhub.skyq_hub import SkyQHub
 import voluptuous as vol
 
 from homeassistant.components.device_tracker import (
     DOMAIN,
     PLATFORM_SCHEMA,
     DeviceScanner,
 )
-from homeassistant.const import CONF_HOST, HTTP_OK
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
-_MAC_REGEX = re.compile(r"(([0-9A-Fa-f]{1,2}\:){5}[0-9A-Fa-f]{1,2})")
+_ERROR = logging.ERROR
+_INFO = logging.INFO
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Optional(CONF_HOST): cv.string})
 
 
-def get_scanner(hass, config):
+async def async_get_scanner(hass, config):
     """Return a Sky Hub scanner if successful."""
-    scanner = SkyHubDeviceScanner(config[DOMAIN])
-
+    host = config[DOMAIN].get(CONF_HOST, "192.168.1.254")
+    websession = async_get_clientsession(hass)
+    hub = SkyQHub(websession, host)
+    scanner = SkyHubDeviceScanner(config[DOMAIN], hub, host)
+    await scanner.async_connect(hass)
     return scanner if scanner.success_init else None
 
 
 class SkyHubDeviceScanner(DeviceScanner):
     """This class queries a Sky Hub router."""
 
-    def __init__(self, config):
+    def __init__(self, config, hub, host):
         """Initialise the scanner."""
-        _LOGGER.info("Initialising Sky Hub")
-        self.host = config.get(CONF_HOST, "192.168.1.254")
+        self._connection_failed = False
+        self._dataparse_failed = False
+        self._hub = hub
+        self.host = host
```

Comment:
Host attribute isn't used.

Reply:
Strange the linting didn't spot that, but vulture has, so removed